src/App.js

```diff
@@ -9,18 +9,22 @@ import ChatWindow from './ChatWindow.js'
 class App extends Component {
   constructor(props) {
     super(props)
-    this.state = {}
+    this.state = {
+      authorization: {}
+    }
   }
 
   render() {
     return (
       <MuiThemeProvider>
         <div className="App">
-          <LoginWithAmazon setAccessToken={(access_token) => this.setState({access_token: access_token})} />
+          <LoginWithAmazon setAuthorization={(authResponse) =>
+            this.setState({authorization: {access_token: authResponse.access_token, expires_in: authResponse.expires_in}})} />
```

Comment:
What if authResponse is undefined?
Read if inline functions is a good practice when they are this long.

Reply:
Moved the function to a handler.